### ISSUE TITLE
support for service workers

### ISSUE THREAD

**Issue by jessewgibbs** [rollbar/rollbar.js#368]
Details in https://app.clubhouse.io/rollbar/story/1641/rollbar-js-support-for-service-workers

**Comment by hexsprite** [comment]
intriguing! but the link doesn't work

**Comment by hexsprite** [comment]
I ended up rolling my own serviceworker support to catch exceptions and log them directly through my application to rollbar

**Comment by rokob** [comment]
The link is to an internal tracking system, not much extra info there anyway. It seems to me that the node side of this library should almost just work inside a service worker, but probably needs a little bit extra for requests.

We should get an example service worker to see where things break down. Although maybe just some documentation to say how to catch your own exceptions and send them over to your application to then send to rollbar might be a good first step.

**Comment by hexsprite** [comment]
this is what I have: https://gist.github.com/hexsprite/852596b9a4d797548ccee7e1b49f2fc1

**Comment by rivkahstandig3636** [comment]
Hi there, I’m closing out all issues opened before 2018 that haven’t had any activity on them since the start of this year. If this is still an issue for you, please comment here and we can reopen this. Thanks!

**Comment by rokob** [comment]
Pretty sure this works on service workers now (at the next release), we can reopen if that is not the case.

**Comment by silentvick** [comment]
Doesn't seem to work. I get this error when trying to impor it in service worker

```
Uncaught ReferenceError: window is not defined
    at eval (webpack:///./node_modules/rollbar/dist/rollbar.umd.min.js?:1:52)
    at Object../node_modules/rollbar/dist/rollbar.umd.min.js (http://localhost:8080/serviceWorker.js:234:1)
    at __webpack_require__ (http://localhost:8080/serviceWorker.js:20:30)
    at eval (webpack:///./src/serviceWorker.js?:3:65)
    at Module../src/serviceWorker.js (http://localhost:8080/serviceWorker.js:384:1)
    at __webpack_require__ (http://localhost:8080/serviceWorker.js:20:30)
    at eval (webpack:///multi_(webpack)-dev-server/client?:2:18)
    at Object.1 (http://localhost:8080/serviceWorker.js:407:1)
    at __webpack_require__ (http://localhost:8080/serviceWorker.js:20:30)
    at http://localhost:8080/serviceWorker.js:84:18
```

Rollbar version 2.7.0

**Comment by brianr** [comment]
thanks @silentvick - re-closing this; we'll track the issue over at #732.

**Comment by brianr** [comment]
@silentvick This regression has been fixed and released in version 2.7.1.
